File: 1. Two Sum/1. Two Sum.py
class Solution:
    """
    This is not good problem. The idea is use hash dict to store position and use O(1) to search.
    """
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = dict()
        for i in range(len(nums)):
            if nums[i] in d:
                return [d[nums[i]], i]
            d[target - nums[i]] = i

File: 1.Two Sum/1.Two Sum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        /*
        same idea with python version
        */
        std::unordered_map<int, int> hash_map;
        std::vector<int> result(2);

        for(int i=0; i<nums.size(); i++)
        {
            int value = target - nums[i];
            if(hash_map.find(value) != hash_map.end())
            {
                result[0] = hash_map[value];
                result[1] = i;
                return result;
            }
            hash_map[nums[i]] = i;

        }
        return result;
    }
};
